tests/test_variant.h

```diff
@@ -105,6 +105,600 @@ TEST_CASE("[Variant] Writer and parser float") {
 
 	CHECK_MESSAGE(b64_float_parsed == 340282001837565597733306976381245063168.0, "Should not overflow.");
 }
+
+TEST_CASE("[Variant] To Bool from Int,Float,String,Vec2,Vec2i,Vec3,Vec3i and Color") {
+	Variant int_v = 0;
+	Variant bool_v = true;
+	int_v = bool_v; // int_v is now a bool
+	CHECK(int_v == Variant(true));
+	bool_v = false;
+	int_v = bool_v;
+	CHECK(int_v == Variant(false));
+
+	Variant float_v = 0.0f;
+	bool_v = true;
+	float_v = bool_v;
+	CHECK(float_v == Variant(true));
+	bool_v = false;
+	float_v = bool_v;
+	CHECK(float_v == Variant(false));
+
+	Variant string_v = "";
+	bool_v = true;
+	string_v = bool_v;
+	CHECK(string_v == Variant(true));
+	bool_v = false;
+	string_v = bool_v;
+	CHECK(string_v == Variant(false));
```

Comment:
```suggestion
	CHECK(string_v == String("False"));
```

And so on, the same applies to the rest of assertions.

If you'd like to see what values get converted to, you can use this snippet in GDScript:
```gdscript
extends Node2D

func _ready():
	var v = convert("", TYPE_BOOL) # Lookup TYPE_* in the documentation for other `Variant` types.
	print(typeof(v))
	print(v)
```

That said, you can also test `Variant::construct()` directly if you'd like to test conversion between various types this way.